walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,67 @@
 
 package com.google.sps;
 
+import java.sql.Time;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
 
 public final class FindMeetingQuery {
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    ArrayList<TimeRange> eRanges = new ArrayList<>();
+    Collection<String> attendees = request.getAttendees();
+    for(Event event: events){
+      Collection<String> eAttendees = event.getAttendees();
+      for(String attendee: attendees){
+        if(eAttendees.contains(attendee)||eAttendees.isEmpty()){
+          eRanges.add(event.getWhen());
+        }
+      }
+    }
+    Collection<TimeRange> overlapedRanges = new ArrayList<>();
+      eRanges.sort(TimeRange.ORDER_BY_START);
+      for(int i = 0; i < eRanges.size();i++){
+        TimeRange currentUnion = eRanges.get(i);
+        if(i+1 < eRanges.size()){          
+          TimeRange next = eRanges.get(i+1);  
+          if(currentUnion.overlaps(next)){
+            if(!currentUnion.contains(next)){
+            currentUnion = TimeRange.fromStartEnd(currentUnion.start(), next.end(), false);
+            }
+            i++;
+          }
+        }
+        overlapedRanges.add(currentUnion);
+      }
+    Collection<TimeRange> validRanges = getValidTimes(overlapedRanges,request.getDuration());
+  return validRanges;
+  }
+
```

Comment:
Can you make your assumptions here a little more clear? You are passing in overlappedRanges above, not eRanges. What are your assumptions on "ranges" below. E.g.:
1) They are sorted by starting time order.
2a) The intervals will have no overlap
2b) Intervals contained inside another interval will be deleted

I am not sure exactly what you are assuming. Currently, it looks like your kind of implements 2b, but you probably want 2a? It seems your code below assumes 2a.